### PR TITLE
plugins/sed: add sid password change

### PR DESCRIPTION
**Pull request by gjoyce-ibm**
The existing ioctl to change password only updated the admin1 password. Add using a new ioctl IOC_OPAL_SET_SID_PW to also update the sid password so that sid and admin1 passwords are kept in sync.

### CODE REVIEW

**Comment by igaw** [comment]
Looks good to me. Let's wait until the matching kernel code has been merged.

**Comment by igaw** [comment]
I might have missed  the kernel change, is this still the last version:

https://lore.kernel.org/all/20240829175639.6478-1-gjoyce@linux.ibm.com

**Comment by gjoyce-ibm** [comment]
> I might have missed the kernel change, is this still the last version:
> 
> https://lore.kernel.org/all/20240829175639.6478-1-gjoyce@linux.ibm.com

Yes. That's the last version. No code changes but I added to the description and added a reference to this pull request.

In three weeks there have been no comments on the resend.

**Comment by igaw** [comment]
I'll bring this up at Linux Plumbers or at Alpss next week. Let's see if this helps.

**Comment by gjoyce-ibm** [comment]
> I'll bring this up at Linux Plumbers or at Alpss next week. Let's see if this helps.

Any discussions?

**Comment by igaw** [comment]
Yes I discussed it with Keith and he was somewhat familiar with the protocol. I was not able to remember the details why this is necessary but he said he didn't really mind the change though, IIRC. 

Let me ping him again here, @keithbusch would you mind to queue up or review these patches: https://lore.kernel.org/all/20240829175639.6478-1-gjoyce@linux.ibm.com

Thanks!

**Review by keithbusch — APPROVED** [review]

**Comment by igaw** [comment]
Upstream kernel patch got applied, so let's apply this one too.

**Pull request merged**
